For the statistical calculation of the t-test(a significant difference in the categories) just for practice, can maybe use both z and t test (should be able to derive the same ans)

In [2]:
import numpy as np
import pandas as pd
from statsmodels.stats import weightstats 
import scipy.stats as stats

In [4]:
transac_data = pd.read_csv('Dummy Data (3 months).csv')

In [5]:
transac_data.head(10)

Date Description of Purchase Category of Purchase  Price 
0   1/1/22           Grocery Store            Groceries   45.67
1   2/1/22             Gas Station       Transportation   35.12
2   3/1/22             Coffee Shop           Dining Out    8.99
3   4/1/22           Movie Theater        Entertainment   22.50
4   5/1/22              Amazon.com      Online Shopping   67.89
5   6/1/22              Hair Salon        Personal Care   55.00
6   7/1/22          Gym Membership       Health/Fitness   80.00
7   8/1/22          Clothing Store             Clothing  112.45
8   9/1/22           Grocery Store            Groceries   23.12
9  10/1/22             Gas Station       Transportation   40.23

In [6]:
#getting rid of whitespace in the headers
transac_data.columns = transac_data.columns.str.strip()

**Analysis 1**: Any significant difference between the amount spent within the different categories

Categories: Groceries, Transportation, Dining Out, Entertainment, Online Shopping, Personal Care 

In [7]:
#filtered item for groceries
groceries = transac_data[transac_data['Category of Purchase']== 'Groceries']['Price']
groceries

0     45.67
8     23.12
16    41.23
24    27.23
30    42.89
38    23.56
46    37.23
54    29.23
58    35.67
66    28.34
74    31.23
82    26.23
Name: Price, dtype: float64

In [8]:
#filtered item for Transportation
transportation = transac_data[transac_data['Category of Purchase']== 'Transportation']['Price']
transportation

1     35.12
9     40.23
17    37.12
25    38.23
31    30.12
39    40.23
47    35.12
55    38.23
59    25.42
67    36.12
75    29.45
83    39.56
Name: Price, dtype: float64

In [9]:
#filtered item for Dining Out
dining_out = transac_data[transac_data['Category of Purchase']== 'Dining Out']['Price']
dining_out

2      8.99
10    12.99
18     9.99
26    10.99
32     8.50
40    12.99
48     9.99
56    10.99
60     7.99
68    10.99
76     8.50
84    12.99
Name: Price, dtype: float64

In [10]:
#filtered item for Entertainment
entertainment = transac_data[transac_data['Category of Purchase']== 'Entertainment']['Price']
entertainment

3     22.50
11    28.75
19    24.50
27    26.75
33    25.00
41    28.50
49    22.75
57    26.75
61    20.50
69    23.75
77    27.50
85    21.75
Name: Price, dtype: float64

In [11]:
#filtered item for Online Shopping
online_shop = transac_data[transac_data['Category of Purchase'] == 'Entertainment']['Price']
online_shop

3     22.50
11    28.75
19    24.50
27    26.75
33    25.00
41    28.50
49    22.75
57    26.75
61    20.50
69    23.75
77    27.50
85    21.75
Name: Price, dtype: float64

In [12]:
#filtered item for Personal Care
personal_care = transac_data[transac_data['Category of Purchase'] == 'Personal Care']['Price']
personal_care

5     55.0
13    67.0
21    62.0
29    71.0
35    65.0
43    72.0
51    60.0
63    62.0
71    75.0
79    68.0
87    80.0
Name: Price, dtype: float64

----------------------------------------------------------------------------------------------------------------------

In [13]:
#conducting t-test for the different categories
#groceries and transportation
stats.ttest_ind(groceries, transportation, equal_var=True)

Ttest_indResult(statistic=-1.062842990461972, pvalue=0.2993811934896691)

p-value = 0.299 > 0.05. We do not reject the null hypothesis. There is no significant difference in the spendings for groceries and transportation.

In [14]:
#groceries and dining out
stats.ttest_ind(groceries, dining_out, equal_var = True)

Ttest_indResult(statistic=9.679605925063399, pvalue=2.175878554643374e-09)

p-value = 2.175878554643374e-09 < 0.05. We reject null hypothesis. There is a significant difference in the spendings for groceries and dining out.

In [23]:
#groceries and entertainment
stats.ttest_ind(groceries, entertainment, equal_var = True)

Ttest_indResult(statistic=3.266894307563918, pvalue=0.0035282759831674008)

p-value = 0.0035283 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for groceries and entertainment.

In [25]:
#groceries and online shopping
stats.ttest_ind(groceries, online_shop, equal_var = True)

Ttest_indResult(statistic=3.266894307563918, pvalue=0.0035282759831674008)

p-value = 0.0035283 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for groceries and entertainment.

In [26]:
#groceries and personal care
stats.ttest_ind(groceries, personal_care, equal_var = True)

Ttest_indResult(statistic=-10.99910646742078, pvalue=3.5653466492395016e-10)

p-value = 3.5653466492395016e-10 < 0.05. We reject the null hypothesis. There is a significant difference in the spendings for groceries and personal care. 

In [28]:
#transportation and dining out
stats.ttest_ind(transportation, dining_out, equal_var = True)

Ttest_indResult(statistic=17.02244760743162, pvalue=3.7574222893432807e-14)

p-value = 3.7574222893432807e-14. We reject the null hypothesis. There is a significant difference in the spendings for transportation and dining out.

In [30]:
#transportation and entertainment
stats.ttest_ind(transportation, entertainment, equal_var = True)

Ttest_indResult(statistic=6.647865962628232, pvalue=1.1029140191760613e-06)

p-value = 1.1029140191760613e-06 < 0.05. We reject the null hypothesis. There is a significant difference in the spendings for transportation and entertainment. 

In [31]:
#transportation and online shopping
stats.ttest_ind(transportation, online_shop, equal_var = True)

Ttest_indResult(statistic=6.647865962628232, pvalue=1.1029140191760613e-06)

p-value = 1.1029140191760613e-06 < 0.05. We reject the null hypothesis. There is a significant difference in the spendings for transportation and online shopping.

In [32]:
#transportation and personal care
stats.ttest_ind(transportation, personal_care, equal_var = True)

Ttest_indResult(statistic=-12.509145549519868, pvalue=3.371108718508917e-11)

p-value = 3.371108718508917e-11 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for transportation and online shopping.

In [33]:
#dining out and entertainment
stats.ttest_ind(dining_out, entertainment, equal_var = True)

Ttest_indResult(statistic=-15.194939077754787, pvalue=3.7872452230425636e-13)

p-value = 3.7872452230425636e-13 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for dining out and entertainment.

In [34]:
#dining out and online shopping
stats.ttest_ind(dining_out, online_shop, equal_var = True)

Ttest_indResult(statistic=-15.194939077754787, pvalue=3.7872452230425636e-13)

p-value = 3.7872452230425636e-13 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for dining out and online shopping.

In [35]:
#dining out and personal care
stats.ttest_ind(dining_out, personal_care, equal_var = True)

Ttest_indResult(statistic=-26.252136052320694, pvalue=1.5331153801767453e-17)

p-value = 1.5331153801767453e-17 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for dining out and personal care.

In [36]:
#entertainment and online shopping
stats.ttest_ind(entertainment, online_shop, equal_var = True)

Ttest_indResult(statistic=0.0, pvalue=1.0)

????

In [38]:
#entertainment and personal care
stats.ttest_ind(entertainment, personal_care, equal_var = True)

Ttest_indResult(statistic=-18.790126657586825, pvalue=1.301376376101303e-14)

p-value = 1.301376376101303e-14 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for entertainment and personal care.

In [39]:
#online shopping and personal care
stats.ttest_ind(online_shop, personal_care, equal_var = True)

Ttest_indResult(statistic=-18.790126657586825, pvalue=1.301376376101303e-14)

p-value = 1.301376376101303e-14 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for entertainment and personal care.

(-18.790126657586825, 9.096202395700598e-79)